dotnet/TF2SA.Tests.Unit/Http/Base/Serialization/ClassStatsDeserializerTests.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.Generic;
+using System;
+using Monad;
+using TF2SA.Common.Models.LogsTF.GameLogModel;
+using TF2SA.Http.Base.Errors;
+using TF2SA.Http.Base.Serialization;
+using Xunit;
+using System.Linq;
+
+namespace TF2SA.Tests.Unit.Http.Base.Serialization;
+
+public class ClassStatsDeserializerTests
```

Comment:
used these test to ensure I covered that the parent classStats was still correct, and the child weapon stats are correct for 3 cases:
- null weaponstats
- old style weapon stats with only the int kills supplied
- new style thorough weaponstats